presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1537,25 +1545,35 @@ private void undoUpdateStatisticsOperations()
 
         private void executeIrreversibleMetastoreOperations()
         {
-            List<String> failedIrreversibleOperationDescriptions = new ArrayList<>();
-            List<Throwable> suppressedExceptions = new ArrayList<>();
-            boolean anySucceeded = false;
+            List<String> failedIrreversibleOperationDescriptions = Collections.synchronizedList(new ArrayList<>());
+            List<Throwable> suppressedExceptions = Collections.synchronizedList(new ArrayList<>());
+            AtomicBoolean anySucceeded = new AtomicBoolean(false);
+
+            ImmutableList.Builder<CompletableFuture<?>> dropFutures = ImmutableList.builder();
             for (IrreversibleMetastoreOperation irreversibleMetastoreOperation : metastoreDeleteOperations) {
-                try {
-                    irreversibleMetastoreOperation.run();
-                    anySucceeded = true;
-                }
-                catch (Throwable t) {
-                    failedIrreversibleOperationDescriptions.add(irreversibleMetastoreOperation.getDescription());
-                    // A limit is needed to avoid having a huge exception object. 5 was chosen arbitrarily.
-                    if (suppressedExceptions.size() < 5) {
-                        suppressedExceptions.add(t);
+                dropFutures.add(CompletableFuture.runAsync(() -> {
+                    try {
+                        irreversibleMetastoreOperation.run();
+                        anySucceeded.set(true);
                     }
-                }
+                    catch (Throwable t) {
+                        failedIrreversibleOperationDescriptions.add(irreversibleMetastoreOperation.getDescription());
```

Comment:
I would just use:
```
synchronized (failedIrreversibleOperationDescriptions) {
  failedIrreversibleOperationDescriptions.add(irreversibleMetastoreOperation.getDescription());
  // A limit is needed to avoid having a huge exception object. 5 was chosen arbitrarily.
  synchronized (suppressedExceptions) {
  if (suppressedExceptions.size() < 5) {
      suppressedExceptions.add(t);
  }
}
```

this seems more consistent and you don't need to use `synchronizedList`

Reply:
done